### DISCUSSION TITLE
Have 2 G6 (or Libre 2) running at the same time

### DISCUSSION THREAD

**Discussion by robertrub** [NightscoutFoundation/xDrip#2859] [Ideas]
Juggluco has an interesting feature for L2 : start a new sensor warm-up while the "old" one is still running. This permits to close the gap between 2 sensors (and also jump over the bumpy readings when a sensor starts).

It would be nice if xDrip could do the same for G6 sensors (and L2 if possible).

**Comment by Navid200** [comment]
It's not xDrip that runs a G6 session.  The session runs on the Dexcom G6 transmitter.
There is nothing we can do in xDrip to eliminate the need to warm up a sensor.

The only option you have for this is to use two G6 transmitters at the same time during the overlap.  This means you will need two active (aging) transmitters considering you cannot deactivate a G6 transmitter after it has been activated.

**Reply by robertrub** [reply]
Yes, 2 transmitters is exactly what I'm saying. 

With Anubis transmitters, which are valid for 180 days, can be restarted, the battery can be changed etc, one can afford to have 2 active transmitters.

If xDrip could handle 2 transmitters at the same time, it would be great.

**Reply by Navid200** [reply]
> If xDrip could handle 2 transmitters at the same time, it would be great.

It can:

1- Install a variant alongside your existing xDrip: https://navid200.github.io/xDrip/docs/Variants.html
2- Establish connectivity to the alternative (2) transmitter using the variant.  
3- When you want to start the parallel session, start it on the variant.  
4- When the parallel session has run enough and is stable enough to your satisfaction and you are ready to switch from your existing session to it, in your main xDrip, change the transmitter ID from 1 to 2.  In the variant, disable collection: https://navid200.github.io/xDrip/docs/Stop-xDrip.html
Your original xDrip will start using the session in progress in a while: https://navid200.github.io/xDrip/docs/Session-in-progress.html  
5- Next time, to start a parallel session, you will need to enable collection in the variant again, but, change the transmitter ID to 1.  

This means you only use the variant when you need to run two sessions in parallel.  And you switch the transmitter ID to go back and forth between transmitters.  The rest of the time, you can disable collection on the variant.
While the variant is active, the main session is on the original xDrip.  So, what is uploaded to Nightscout only comes from the original.  So, you don't need to set up Nightscout in the variant.

**Reply by Navid200** [reply]
I know that's not exactly what you meant.

If we were to implement, in xDrip, the ability to collect from two different transmitters simultaneously, we would need to spend a lot of time and complicate the code.
All of this to address a need that only a small fraction of the user community has.

What I have presented is a work-around.  It accomplishes exactly the objective.  Not as easily as if xDrip could collect from two transmitters.  But, it accomplishes the objective.
Whenever there is a reasonable workaround, we need to focus on other needs including bug fixes.  I hope this is reasonable.

**Reply by robertrub** [reply]
Thanks a lot for the workaround :)

It is not as simple as having in one app just a "switch" to change between 2 transmitters but I suppose your solution is easy enough to use. Will try it. Actually I was playing with xDrip and BYODA to have 2 different apps.

I understand totally the need to use the programming time efficiently :)

**Reply by Magu70** [reply]
So glad to have come across this work around! One doubt of mine--we use fsl 1 + bubble (only option in our country). Calibration is key, of course. Will the calibrations we put in the variant, while warming up the new sensor, move over to the main Xdrip when we make the switch?

**Reply by Navid200** [reply]
No, I am not aware of an existing simple method for moving calibrations from one variant to another.

In general, calibration includes a time parameter.  Every calibration point has a time.  As it ages, it will have less impact.  Even if you have a perfectly calibrated setup now, there is no guarantee it will still be perfect tomorrow.

G6 is an evolution of CGM, which includes factory calibration, which includes the time parameter.

**Reply by robertrub** [reply]
Sorry but I don't think so. The Libre 1 is "dumb" and the software does all "tweaking".
When I was using FSL1 (with Miaomiao), I got the best results with GLIMP (on Play Store) and calibrations. No idea how you can run 2 in parallel :(
Juggluco can run 2 FSL2 or FSL3 in parallel but I don't know if it works with L1. Check 
https://www.juggluco.nl/Juggluco/mgdL/index.html

**Reply by Magu70** [reply]
Thanks to both of you. We'll keep messing with it! Dexcom 1 just arrived to Argentina, but no idea how long until we'll have access to it. But who knows, maybe soon we'll get to dance a new dance. Until then, no complaints. Our set up is a hassle, but we have nothing to compare it to and life is pretty good all the same. =) About to celebrate 3 years of looping with AAPS, so clearly it's possible.

**Reply by Magu70** [reply]
So it appears Juggluco can work with FSL 1s, but no manual calibrations. We really need those for looping. The factory calibration is always way, way off. Right now just toggling between main Xdrip and variant works, except we lose her MiBand connection on the variant. However, in a few weeks will have a galaxy watch that connects to AAPS, so that's fixed.

**Reply by Navid200** [reply]
Why do you need miband connection to the variant?

If you only use the variant to start a new sensor while the existing sensor is already working and miband is receiving from it , why would you ever need miband to receive from the variant?

**Reply by Magu70** [reply]
I have to stay with the version of xdrip i used for starting/calibrating the new sensor. I can't switch transmitters in main xdrip b/c i lose calibrations i did in the variant. For example, to test I went back to xdrip+ just now, that DOES NOT have current calibrations, and values were off by over 100. We calibrate new fsl in first 24 hours, getting 3 good calibrations (stable bg) in the 80s, 100 or so, and 150 or so. With this, we get very stable behavior. Sometimes around day 10 or so we need to adjust a few calibrations, but never do it all over again. Has worked for us for now.

**Reply by Magu70** [reply]
The GOOD NEWS is we're excited to be able to overlap the starting of sensors like this. Right now, the down side is that in the dev version of NS, the display from Xdrip variant has like 20 digits. Can confirm it's the Xdrip variant broadcast that is causing this. (I switched back to xdrip+ and BG display was normal.) The other down side is losing the Miband on variant. HOWEVER, soon we'll have a galaxy watch and we'll connect it strait to aaps. So in the balance, all is good! (The NS display a bit of a hassle, but there are worse things...)

**Reply by Navid200** [reply]
@Magu70 

OK,

I cannot make any promises.
But, if you want, you can open an issue titled "Allow calibrations to be transferred to/from xDrip variants".  
In it, you should explain what you explained here so that any developer can easily see why you cannot use the variant just for starting the new sensor and you have no choice but to completely switch between main and variant.

The facts are: 
I am not sure how easy it is to add this feature.
I am not sure if this can already be done.
I have no idea if this has been intentionally left out for a good reason.

**Reply by Magu70** [reply]
Thank you so much. Really appreciated! Will do.
